src/Bindings/ManualBindings.cpp

```diff
@@ -2775,6 +2777,53 @@ static int tolua_cCompositeChat_UnderlineUrls(lua_State * tolua_S)
 
 
 
+static int tolua_cChannelManager_RegisterChannel(lua_State * tolua_S)
+{
+	
+	// Retrieve the cPlugin from the LuaState:
+	cPluginLua * Plugin = cManualBindings::GetLuaPlugin(tolua_S);
+	if (Plugin == nullptr)
+	{
+		// An error message has been already printed in GetLuaPlugin()
+		return 0;
+	}
+	
+	// Check params:
+	cLuaState L(tolua_S);
+	if (!L.CheckParamUserType(1, "cChannelManager"))
+	{
+		return 0;
+	}
+	cChannelManager * self = reinterpret_cast<cChannelManager *>(tolua_tousertype(tolua_S, 1, nullptr));
+	if (self == nullptr)
+	{
+		return cManualBindings::lua_do_error(tolua_S, "invalid 'self' in function 'cCompositeChat:UnderlineUrls'");
+	}
+	if (!L.CheckParamString(2) || !L.CheckParamFunction(3))
+	{
+		return 0;
+	}
+
+	AString Channel;
+	L.GetStackValue(2, Channel);
+	int FnRef = luaL_ref(tolua_S, LUA_REGISTRYINDEX);
+	if (FnRef == LUA_REFNIL)
+	{
+		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #3");
+	}
+	auto Callback = std::make_shared<cChannelCallback>(*Plugin, FnRef);
+
+	self->RegisterChannel(Channel, Callback);
+	
+	// Cut away everything from the stack except for the cChannelManager reference; return that:
+	lua_settop(L, 1);
```

Comment:
No, ignore my comment, I'm getting rusty in my C++-Lua bindings knowledge. Sorry.